### ISSUE TITLE
Chains should sign and attest to all Image Manifests in an Image Index

### ISSUE THREAD

**Issue by arewm** [tektoncd/chains#1070]
### Feature request

If Chains is provided with an Image Index to sign and attest, it should recursively perform this same behavior for all referenced Image Manfiests as well.

### Use case

In order to improve the experience for increasing supported architectures for images, some build tasks may choose to always produce Image Index OCI artifacts even if there is only a single architecture referenced. As architectures are added to the Image Index, the Image Manifests should be signed without requiring that the specific pullspecs are included as results on the pipeline.

**Comment by lcarva** [comment]
+1

This was discussed in check a few days ago. The only concern raised was that this behavior should be behind a flag, at least initially.

**Comment by lcarva** [comment]
It does make sense. I'd like to see this behavior in cosign itself. The CLI does have a `recursive` flag, but I'm not sure if this will already handle truly recursive Image Indexes or if it needs some tweaking.

**Comment by arewm** [comment]
While I think that this is still beneficial functionality, it should be opt-in. There may be a task which cannot claim that one of the referenced manifests are build in that task (i.e. it consumed one manifest, produced another, and wrapped both in an index).

**Comment by chmeliik** [comment]
> While I think that this is still beneficial functionality, it should be opt-in. There may be a task which cannot claim that one of the referenced manifests are build in that task (i.e. it consumed one manifest, produced another, and wrapped both in an index).

Ideally opt-in on the Task / TaskRun level? That would allow the Task author to decide if they want the behavior to be recursive or not.

**Comment by arewm** [comment]
Yes, at the TaskRun level. I imagine that it should also be configurable based on the results... but maybe it would just be better to have one result that is recursive and another that isn't?
